core/camel-support/src/main/java/org/apache/camel/support/builder/TokenXMLExpressionIterator.java

```diff
@@ -239,7 +239,31 @@ String getNext(boolean first) {
             
             return next;
         }
-
+        
+        private String getMissingInherritNamespaces(final String text) {
+            if (text == null) {
+                return "";
+            }
+            final String namespaces = getNamespacesFromNamespaceToken(text);
+            final String[] containedNamespaces = namespaces == null ? new String[0] : namespaces.split(" ");
+            final StringBuilder sb = new StringBuilder();
```

Comment:
You can use CollectionStringBuilder from Camel to add a bunch of stuff together with a separator and it deals with the "first" issue etc.

Reply:
The CollectionStringBuilder wont work here because if already a part of multiple childnamespaces are available it is required to add a separator even it is the first match. 

e.g. Input
```
<orders xmlns="http:acme.com" xmlns:foo="http:foo.com">
  <order id="1" xmlns="http:acme.com">Camel in Action</order>
</orders>
```

will be with CollectionStringBuffer without separation
```
  <order id="1" xmlns="http:acme.com"xmlns:foo="http:foo.com">Camel in Action</order>
```
but should be with a additional separator
```
  <order id="1" xmlns="http:acme.com"xmlns:foo="http:foo.com">Camel in Action</order>
```